### PR TITLE
JAMES-2949 Post refactoring cleanup: Remove JamesUser

### PR DESCRIPTION
**Pull request by chibenwa**
Specific handling, as well as this model, is thus no longer required, after
removal of JamesUsersRepository & UsersFileRepository

See #2843

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[5ea90878e9076f488e26815f505a20fc58ead3f1] [ERROR] Failed to execute goal on project james-json: Could not resolve dependencies for project org.apache.james:james-json:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.9.9 -> com.google.guava:guava:jar:25.1-jre -> com.google.errorprone:error_prone_annotations:jar:2.1.3: Failed to read artifact descriptor for com.google.errorprone:error_prone_annotations:jar:2.1.3: Could not transfer artifact com.google.errorprone:error_prone_annotations:pom:2.1.3 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.pom.part (No such file or directory) -> [Help 1]
```

test this please

**Comment by Arsnael** [comment]
```
[ERROR] Failed to execute goal on project event-sourcing-pojo: Could not resolve dependencies for project org.apache.james:event-sourcing-pojo:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre -> org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Failed to read artifact descriptor for org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Could not transfer artifact org.codehaus.mojo:animal-sniffer-annotations:pom:1.14 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.pom.part (No such file or directory) -> [Help 1]
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`MemorySpamAssassinContractTest.expungingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout`

test this please

**Comment by chibenwa** [comment]
Merged

File: server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepositoryManagementMBean.java [hunk 1]
```diff
@@ -91,55 +89,6 @@
...
-     * Removes a user's alias which terminates local mail forwarding
-     * 
-     * 
-     * @param userName
-     *            The name of the user whose alias is unset
-     * @throws UsersRepositoryException
-     *             if error
-     */
-    @Deprecated
-    void unsetAlias(Username userName) throws Exception;
```
Comment (mbaechler) [inline server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepositoryManagementMBean.java]: as long as we will support JMX, all MBean classes should be considered a public API of James.
Removing and/or changing them should be considered an api breakage.

Comment (chibenwa) [reply server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepositoryManagementMBean.java]: We can still keep the endpoint (akka the API methods), and throw unsupported exceptions as the previous code always ended up throwing IllegalArgumentExceptions....

Would that be acceptable to you?

Comment (chibenwa) [reply server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepositoryManagementMBean.java]: See https://github.com/linagora/james-project/pull/2924 for announcing that breaking change.

**Pull request closed without merging**